### ISSUE TITLE
Bert Dropout+Add Example

### ISSUE THREAD

**Issue by kevinstephano** [csarofeen/pytorch#472]
## 🐛 Bug

This isn't really a bug, just posting a script and results.

## To Reproduce

You need to cherry pick Jie's branch.

```
git cherry-pick 097ad454a7d9f04935909d89ab256f2baa532862
```

Script for Dropout-Add

```
import torch.nn.functional as F
import torch.nn as nn
import torch

torch._C._jit_set_nvfuser_enabled(True)
torch._C._jit_set_texpr_fuser_enabled(False)                                                                                                                                                 
torch._C._jit_set_profiling_executor(True)                                                                                                                                              
torch._C._jit_set_profiling_mode(True)                                                                                                                                                  
torch._C._jit_override_can_fuse_on_cpu(False)                                                                                                                                           
torch._C._jit_override_can_fuse_on_gpu(False)                                                                                                                                           
torch._C._jit_set_bailout_depth(20)                                                                                                                                                     

class BertOutput(nn.Module):
    def __init__(self, hidden_size, intermediate_size, hidden_dropout_prob, layer_norm_eps=1e-05):
        super().__init__()
        self.dense = nn.Linear(intermediate_size, hidden_size)
        self.LayerNorm = nn.LayerNorm(hidden_size, eps=layer_norm_eps)
        self.dropout = nn.Dropout(hidden_dropout_prob)

    def forward(self, hidden_states, input_tensor):
        hidden_states = self.dense(hidden_states)
        hidden_states = self.dropout(hidden_states)
        hidden_states = self.LayerNorm(hidden_states + input_tensor)
        return hidden_states


model = BertOutput(768, 3072, 0.1)
model.to('cuda')

jit_model = torch.jit.script(model)

inputs = torch.randn([32, 128, 3072], device='cuda', requires_grad=True)
resids = torch.randn([32, 128, 768], device='cuda', requires_grad=True)

for idx in range(0, 5) :
    if idx == 1 :
        print(jit_model.forward.graph_for(inputs, resids))
    jit_model.forward(inputs, resids)
```

## Expected behavior

```
graph(%self : __torch__.BertOutput,                                                                                                                                                                                                                                    [78/1875]
      %hidden_states.1 : Tensor,                                  
      %input_tensor.1 : Tensor):              
  %8 : float = prim::Constant[value=1.0000000000000001e-05]() # /opt/conda/lib/python3.8/site-packages/torch/nn/modules/normalization.py:170:66
  %7 : bool = prim::Constant[value=1]() # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:2072:28
  %5 : int = prim::Constant[value=2]() # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:1663:22
  %4 : int = prim::Constant[value=1]()                             
  %3 : int[] = prim::Constant[value=[768]]()             
  %9 : __torch__.torch.nn.modules.linear.Linear = prim::GetAttr[name="dense"](%self)
  %10 : Tensor = prim::GetAttr[name="weight"](%9)                                                                            
  %11 : Tensor = prim::GetAttr[name="bias"](%9)            
  %13 : int = aten::dim(%hidden_states.1) # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:1663:7
  %14 : bool = aten::eq(%13, %5) # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:1663:7
  %hidden_states.3 : Tensor = prim::If(%14) # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:1663:4
    block0():
      %53 : Tensor = prim::DifferentiableGraph_0(%11, %hidden_states.1, %10)
      -> (%53)
    block1():
      %72 : Tensor = aten::t(%10) # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:1667:30
      %output.2 : Tensor = aten::matmul(%hidden_states.1, %72) # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:1667:17
      %output.3 : Tensor = aten::add_(%output.2, %11, %4) # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:1669:12
      -> (%output.3)
  %30 : __torch__.torch.nn.modules.dropout.Dropout = prim::GetAttr[name="dropout"](%self)
  %31 : bool = prim::GetAttr[name="training"](%30) 
  %33 : __torch__.torch.nn.modules.normalization.LayerNorm = prim::GetAttr[name="LayerNorm"](%self)
  %37 : Tensor = prim::GetAttr[name="weight"](%33)
  %38 : Tensor = prim::GetAttr[name="bias"](%33)
  %45 : Tensor = prim::DifferentiableGraph_1(%input_tensor.1, %hidden_states.3, %31)
  %hidden_states.7 : Tensor = aten::layer_norm(%45, %3, %37, %38, %8, %7) # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:2071:11
  return (%hidden_states.7) 
with prim::DifferentiableGraph_0 = graph(%8 : Tensor,
      %15 : Tensor,
      %19 : Tensor):
  %20 : Tensor = prim::profile[profiled_type=Tensor](%19)
  %18 : Tensor = aten::t(%20) # /opt/conda/lib/python3.8/site-packages/torch/nn/functional.py:1665:39
  %16 : Tensor = prim::profile[profiled_type=Tensor](%15)
  %14 : Tensor = prim::profile[profiled_type=Tensor](%18)
  %12 : Tensor = aten::mm(%16, %14) # <string>:3:24
  %9 : Tensor = prim::profile[profiled_type=Tensor](%8)
  %7 : Tensor = prim::profile[profiled_type=Tensor](%12)
  %4 : int = prim::Constant[value=1]()
  %5 : Tensor = aten::add(%9, %7, %4) # <string>:3:17
  %68 : int[] = aten::size(%9) # <string>:3:44
  %69 : int[] = aten::size(%5) # <string>:3:55
  %70 : int[]? = aten::_size_if_not_equal(%68, %69) # <string>:3:19
  %71 : int[] = aten::size(%7) # <string>:3:93 
  %73 : int[]? = aten::_size_if_not_equal(%71, %69) # <string>:3:68
  %74 : (int[]?, int[]?) = prim::TupleConstruct(%70, %73)
  %self_size.3 : int[]?, %other_size.3 : int[]? = prim::TupleUnpack(%74)
  %1 : Tensor = prim::profile[profiled_type=Tensor](%5)
  return (%1, %16, %14, %self_size.3, %other_size.3)
with prim::DifferentiableGraph_1 = graph(%6 : Tensor,
      %10 : Tensor,
      %12 : bool):
  %11 : float = prim::Constant[value=0.10000000000000001]()
  %37 : None = prim::Constant()
  %38 : float = prim::Constant[value=1.]() # <string>:100:18
  %39 : int = prim::Constant[value=1]() # <string>:103:56
  %p1m.3 : float = aten::sub(%38, %11) # <string>:100:18
  %41 : Tensor = aten::rand_like(%10, %37, %37, %37, %37, %39) # <string>:103:19
  %mask.3 : Tensor = aten::lt(%41, %p1m.3) # <string>:103:19
  %43 : Tensor = aten::type_as(%mask.3, %10) # <string>:104:18
  %44 : Tensor = aten::mul(%43, %10) # <string>:104:18
  %45 : float = aten::div(%38, %p1m.3) # <string>:104:49
  %res.1 : Tensor = aten::mul(%44, %45) # <string>:104:18
  %9 : Tensor = prim::profile[profiled_type=Float(32, 128, 768, strides=[98304, 768, 1], requires_grad=0, device=cuda:0)](%res.1)
  %7 : Tensor = prim::profile[profiled_type=Float(32, 128, 768, strides=[98304, 768, 1], requires_grad=0, device=cuda:0)](%6)
  %5 : Tensor = aten::add(%9, %7, %39) # bert_dropout_add.py:23:39
  %58 : int[] = aten::size(%9) # <string>:3:44
  %59 : int[] = aten::size(%5) # <string>:3:55
  %60 : int[]? = aten::_size_if_not_equal(%58, %59) # <string>:3:19
  %61 : int[] = aten::size(%7) # <string>:3:93 
  %63 : int[]? = aten::_size_if_not_equal(%61, %59) # <string>:3:68
  %64 : (int[]?, int[]?) = prim::TupleConstruct(%60, %63)
  %self_size.3 : int[]?, %other_size.3 : int[]? = prim::TupleUnpack(%64)
  %1 : Tensor = prim::profile[profiled_type=Float(32, 128, 768, strides=[98304, 768, 1], requires_grad=0, device=cuda:0)](%5)
  return (%1, %p1m.3, %mask.3, %self_size.3, %other_size.3)
```

```
__global__ void kernel1(Tensor<float, 3> T0, Tensor<float, 3> T1, Tensor<float, 3> T8, Tensor<bool, 3> T4, unsigned long long seed, unsigned long long offset) {
  const int idx = blockIdx.x*blockDim.x + threadIdx.x;
  Philox rnd(seed, idx, offset);
  float T2[1];
  float T5[1];
  float T6[1];
  float T7[1];
  if ((((((blockIdx.x * 1) + (1 - 1)) * 128) + threadIdx.x) < ((T1.size[0] * T1.size[1]) * T1.size[2]))) {
    for(size_t ki58 = 0; ki58 < 1; ++ki58) {
      T2[ki58]
         = randLike(rnd);
      T4[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)]
        = T2[ki58]
        < float(0.89999997615814209);
      T5[ki58]
         = float(T4[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)]);
      T6[ki58]
        = T5[ki58]
        * T1[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)];
      T7[ki58]
        = T6[ki58]
        * float(1.1111111640930176);
      T8[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)]
        = T7[ki58]
        + T0[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)];
    }
  } else {
    for(size_t ki58 = 0; ki58 < 1; ++ki58) {
      if ((((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x) < ((T1.size[0] * T1.size[1]) * T1.size[2]))) {
        T2[ki58]
           = randLike(rnd);
      }
      if ((((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x) < ((T1.size[0] * T1.size[1]) * T1.size[2]))) {
        T4[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)]
          = T2[ki58]
          < float(0.89999997615814209);
      }
      if ((((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x) < ((T1.size[0] * T1.size[1]) * T1.size[2]))) {
        T5[ki58]
           = float(T4[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)]);
      }
      if ((((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x) < ((T1.size[0] * T1.size[1]) * T1.size[2]))) {
        T6[ki58]
          = T5[ki58]
          * T1[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)];
      }
      if ((((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x) < ((T1.size[0] * T1.size[1]) * T1.size[2]))) {
        T7[ki58]
          = T6[ki58]
          * float(1.1111111640930176);
      }
      if ((((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x) < ((T1.size[0] * T1.size[1]) * T1.size[2]))) {
        T8[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)]
          = T7[ki58]
          + T0[((((blockIdx.x * 1) + ki58) * 128) + threadIdx.x)];
      }
    }
  }
}
```

**Comment by csarofeen** [comment]
Closing, please reopen if there's an issue.
